manifests/job/present.pp

```diff
@@ -5,7 +5,10 @@
 # Parameters:
 #
 #   config
-#     the content of the jenkins job config file (required)
+#     the content of the jenkins job config file (template based)
+#
+#   temp_config_path
```

Comment:
If I'm understanding the problem correctly here, the goal is to make the intermediate file used for configuring the job able to be overridden right?

If that's the case, I'm trying to think of a parameter name that would make that more clear so a user wouldn't be confused as to the purpose of the parameter, without knowledge of how the defined type works internally :/